### ISSUE TITLE
RFC: Renovate the `remote-loader` (or deprecate)

### ISSUE THREAD

**Issue by andrii-bodnar** [lingui/js-lingui#1551]
[@lingui/remote-loader](https://www.npmjs.com/package/@lingui/remote-loader) is a package that allows loading and compiling message catalogs located somewhere remotely (for example, an external CDN endpoint).

For some reason, it was marked as `private` and the latest published version is [3.11.0](https://www.npmjs.com/package/@lingui/remote-loader/v/3.11.0).

Today, almost every Localization Management System has an Over-The-Air Content Delivery feature that allows translations to be distributed to a CDN (or other resources located on the web) and then applications can consume these translations directly from the remote endpoint. For example, this is how it works in Crowdin - [Content Delivery](https://support.crowdin.com/content-delivery/).

In short, OTA Content Delivery allows new translations to be delivered without necessarily rebuilding or redeploying the application. So the Remote Loader would be a good complement to OTA technologies.

**Possible steps to do**:

- [ ] Review the codebase, actualize it
- [ ] Add docs for the package
- [ ] Update and fix [README.md](https://github.com/lingui/js-lingui/blob/main/packages/remote-loader/README.md)
- [ ] Make it public

**In case if it's possible just with the** [@lingui/loader](https://www.npmjs.com/package/@lingui/loader):

- [ ] Add the corresponding section to the [`@lingui/loader` docs](https://lingui.dev/ref/loader) describing how to load resources from the remote host
- [ ] Remove the `remote-loader` package from the v4 codebase
- [ ] Deprecate the package on NPM

**Comment by thekip** [comment]
Currently, the only one problem here, it doesn't support any other formats except of `minimal`.  To make it work with all formats, custom formatters should be integrated in this loader. 

By the way, Pofile would be tricky to bundle for for the browser because it has `require('fs')`, one more point for adopting and maintaining https://github.com/thekip/pofile under lingui org

**Comment by andrii-bodnar** [comment]
I wonder if this should be considered every time someone creates a custom formatter. What about other formatters? Are they compatible with this?

**Comment by thekip** [comment]
Yes, it should be considered when someone creates a custom formatter. Formatters shouldn't have any browser-incompatibile dependencies.  Other official lingui formatters are compatible with browser.

**Comment by andrii-bodnar** [comment]
@thekip I just thought if Pofile is a blocker in the current case, we probably can adopt its fork under the Lingui organization

**Comment by andrii-bodnar** [comment]
Now Crowdin can provide already compiled translations via OTA CDN - [Lingui String Exporter](https://store.crowdin.com/lingui-string-exporter)
